components/calendar/src/any_calendar.rs

```diff
@@ -279,7 +287,9 @@ impl AnyCalendar {
 
     /// Constructs an AnyCalendar for a given calendar kind and [`BufferProvider`] data source
     ///
-    /// For calendars that need data, will attempt to load the appropriate data from the source
+    /// For calendars that need data, will attempt to load the appropriate data from the source.
+    ///
+    /// This API needs the `calendar/japanese@1` data key if working with Japanese calendars.
```

Comment:
Suggestion (pre-existing code): You should be able to make `_with_any_provider` and `_with_buffer_provider` both delegate to `_unstable`.